library/std/src/thread/spawnhook.rs

```diff
@@ -0,0 +1,145 @@
+use crate::cell::Cell;
+use crate::sync::Arc;
+use crate::thread::Thread;
+
+// A thread local linked list of spawn hooks.
+crate::thread_local! {
+    static SPAWN_HOOKS: Cell<SpawnHooks> = const { Cell::new(SpawnHooks { first: None }) };
```

Comment:
Why does this use a linked list rather than a Vec?

Reply:
So the Vec doesn't have to be cloned when a new thread is spawned. It just creates an extra Arc to the (current) head of the list, making it a linked 'tree' with multiple heads.